app/javascript/components/applications/application_form/AddressForm.js

```diff
@@ -1,20 +1,20 @@
 import React from 'react'
-import { Text } from 'react-form'
+import { FieldWrapper } from '~/utils/form/final_form/Field'
 import { isEmpty } from 'lodash'
 import { maxLengthMap } from '~/utils/formUtils'
 
-const buildField = (memberType, nestedField, fieldMap, fieldKey) => {
+const buildField = (memberType, name, fieldMap, fieldKey) => {
   // to do: refactor this to use props
   if (memberType === 'primaryApplicant') {
-    return fieldMap[fieldKey]
-  } else if (memberType === 'householdMember') {
```

Comment:
@akegan looks like we used to expect "householdMember" as a value for `memberType` here, but now we don't - what's the context around that change?

Reply:
I didn't make this change, but it seems like it was an update to again accomodate the field array syntax. Maybe we should actually put this first, so the logic is: use the name if you have it, and fall back to applicant or alternate_contact if the name isn't defined.

@software-project do you have any more context?

Reply:
This is due to different way that forms are working with final form. HouseholdMembers are using FormArray. But in fact I can simplify it further.

Reply:
@software-project Ok that's a lot simpler! Could we also differentiate the var names `fieldName`, `fieldMap`, and `fieldKey` a little bit more? Seems like the field in `fieldName` is like a level higher than the field in `fieldMap` and `fieldKey`. If that's the case, it could be helpful to indicate that in the var names. Does something like `subFieldMap` and `subFieldKey` make sense? Or `addressFieldMap` and `addressFieldKey`? Something to show that there's this parent field with nested address fields.